runtime/pack/dist/opt/comment/autoload/comment.vim

```diff
@@ -76,3 +76,84 @@ export def Toggle(...args: list<string>): string
     noautocmd keepjumps setline(lnum1, lines)
     return ''
 enddef
+
+
+# Comment text object
+# Usage:
+#     import autoload 'dist/comment.vim'
+#     onoremap <silent>ic <scriptcmd>comment.ObjComment(v:true)<CR>
+#     onoremap <silent>ac <scriptcmd>comment.ObjComment(v:false)<CR>
+#     xnoremap <silent>ic <esc><scriptcmd>comment.ObjComment(v:true)<CR>
+#     xnoremap <silent>ac <esc><scriptcmd>comment.ObjComment(v:false)<CR>
+export def ObjComment(inner: bool)
+    def IsComment(): bool
+        var stx = map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')->join()
+        if stx =~ 'Comment'
+            return true
+        else
+            return false
+        endif
+    enddef
+
+    var pos_init = getcurpos()
+
+    # If not in comment, search next one,
+    if !IsComment()
+        if search('\S\+', 'W', line(".") + 100, 100, () => !IsComment()) <= 0
+            return
+        endif
+    endif
+
+    # Search for the beginning of the comment block
+    if IsComment()
+        if search('\v(\S+)|$', 'bW', 0, 200, IsComment) > 0
+            search('\v(\S+)|$', 'W', 0, 200, () => !IsComment())
+        else
+            cursor(1, 1)
+            search('\S\+', 'cW', 0, 200)
+        endif
+    endif
+
+    var pos_start = getcurpos()
+
+    if !inner
+        if search('\s*', 'bW', line('.'), 200) > 0
```

Comment:
That matches very often, even zero times. Can we make this `\s` instead or do we need `\s\+`?

Reply:
right `\s\+`

Reply:
Nah, we can't use search here, will be using while loop instead, example case:

![image](https://github.com/user-attachments/assets/12ecef11-6310-463f-9dd2-3e73e21b1fd1)

```

Reply:
Added test case as well.

Reply:
```
    if !inner
        var col = pos_start[2]
        var prefix = getline(pos_start[1])[ : col - 2]
        while col > 0 && prefix[col - 2] =~ '\s'
            col -= 1
        endwhile
        pos_start[2] = col
        # instead of this:
        # if search('\v\s+', 'bW', line('.'), 200) > 0
        #     pos_start = getcurpos()
        # endif
    endif
```